iceoryx_posh/source/runtime/ipc_runtime_interface.cpp

```diff
@@ -235,19 +235,39 @@ IpcRuntimeInterface::RegAckResult IpcRuntimeInterface::waitForRegAck(int64_t tra
                 }
 
                 // read out the shared memory base address and save it
-                iox::convert::fromString(receiveBuffer.getElementAtIndex(1U).c_str(), m_shmTopicSize);
                 UntypedRelativePointer::offset_t segmentManagerOffset{UntypedRelativePointer::NULL_POINTER_OFFSET};
-                iox::convert::fromString(receiveBuffer.getElementAtIndex(2U).c_str(), segmentManagerOffset);
+                UntypedRelativePointer::offset_t heartbeatOffset{UntypedRelativePointer::NULL_POINTER_OFFSET};
+                int64_t receivedTimestamp{0U};
+
+                auto topic_size_result =
+                    iox::convert::from_string<uint64_t>(receiveBuffer.getElementAtIndex(1U).c_str());
+                auto segment_manager_offset_result =
+                    iox::convert::from_string<uintptr_t>(receiveBuffer.getElementAtIndex(2U).c_str());
+                auto recv_timestamp_result =
+                    iox::convert::from_string<int64_t>(receiveBuffer.getElementAtIndex(3U).c_str());
+                auto segment_id_result =
+                    iox::convert::from_string<uint64_t>(receiveBuffer.getElementAtIndex(4U).c_str());
+                auto heartbeat_offset_result =
+                    iox::convert::from_string<uintptr_t>(receiveBuffer.getElementAtIndex(5U).c_str());
+
+                // validate conversion results
+                if (!topic_size_result.has_value() || !segment_manager_offset_result.has_value()
+                    || !recv_timestamp_result.has_value() || !segment_id_result.has_value()
+                    || !heartbeat_offset_result.has_value())
+                {
+                    return RegAckResult::CONVERSION_FAILURE;
```

Comment:
Food for thought. What do you think of `MALFORMED_RESPONSE` or `CORRUPTED_RESPONSE`?

Oh, and thumbs up for improving this part of the code :)

Reply:
I prefer the former one :)

Reply:
See [809dddc](https://github.com/eclipse-iceoryx/iceoryx/pull/2135/commits/809dddca229ad98fa46ad08ccfd271927ae83cab).